### PR TITLE
Panzer MiniEM: Refactor interpolations, assemble fewer operators

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/panzer 

## Motivation
- Improve assembly of interpolation operators such as discrete gradient and curl and transfers for p-multigrid.
- Skip assembly of operators that are just scalar multiples of each other.

### CODE REVIEW

File: packages/panzer/mini-em/src/solvers/MiniEM_Interpolation.cpp [hunk 1]
```diff
@@ -1,7 +1,74 @@
...
+        auto I   = rowptr(rlid);
+        for (LocalOrdinal k = 0; k < row.length; ++k) {
+          if (impl_ATS::magnitude(row.value(k)) > tol) {
+            idx(I)  = row.colidx(k);
+            vals(I) = row.value(k);
+            I += 1;
+          }
+        }
+      });
+
```
Comment (rppawlo) [inline packages/panzer/mini-em/src/solvers/MiniEM_Interpolation.cpp:63]: perhaps fence here?

**Review by rppawlo — DISMISSED** [review]
LGTM! Thanks @cgcgcg !

File: packages/panzer/mini-em/src/solvers/MiniEM_Interpolation.cpp (hunk 1 above)
Comment (cgcgcg) [reply packages/panzer/mini-em/src/solvers/MiniEM_Interpolation.cpp:64]: Done!

**Review by rppawlo — APPROVED** [review]

**Comment by cgcgcg** [comment]
Thanks @rppawlo !

**Pull request merged**